homeassistant/components/switch/arduino.py

```diff
@@ -28,24 +28,34 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     pins = config.get('pins')
     for pinnum, pin in pins.items():
         if pin.get('name'):
-            switches.append(ArduinoSwitch(pin.get('name'),
-                                          pinnum,
-                                          pin.get('type')))
+            switches.append(ArduinoSwitch(pinnum, pin))
     add_devices(switches)
 
 
 class ArduinoSwitch(SwitchDevice):
     """Representation of an Arduino switch."""
 
-    def __init__(self, name, pin, pin_type):
+    def __init__(self, pin, options):
         """Initialize the Pin."""
         self._pin = pin
-        self._name = name or DEVICE_DEFAULT_NAME
-        self.pin_type = pin_type
+        self._name = options.get('name') or DEVICE_DEFAULT_NAME
+        self.pin_type = options.get('type')
         self.direction = 'out'
-        self._state = False
+
+        default = options.get('default', 'no')
```

Comment:
YAML will actively convert a lot of different strings to booleans, including 'on'. See the regexp [here](http://yaml.org/type/bool.html).

Can you also use `initial` instead of `default` as it better represents what the value will be used for.

```python
initial = options.get('initial', False)
```